CesiumGltf/src/Ktx2TranscodeTargets.cpp

```diff
@@ -0,0 +1,120 @@
+
+#include "CesiumGltf/Ktx2TranscodeTargets.h"
+
+#include <algorithm>
+
+using namespace CesiumGltf;
+
+Ktx2TranscodeTargets::Ktx2TranscodeTargets(
+    const SupportedGpuCompressedPixelFormats& supportedFormats) {
+
+  // Attempts to determine ideal transcode target formats using the logic here:
+  // https://github.com/KhronosGroup/3D-Formats-Guidelines/blob/main/KTXDeveloperGuide.md
+
+  // Find transcode target for ETC1S RGBA
+  if (supportedFormats.ETC2_RGBA) {
+    this->ETC1S_RGBA = GpuCompressedPixelFormat::ETC2_RGBA;
+    //  } else if (suportedFormats.ETC1_RGB) {
+    // It is recommended to transcode into two separate etc1_rgb images here,
+    // but we'll just skip the hassle now and consider bc7 instead.
+  } else if (supportedFormats.BC7_RGBA) {
+    this->ETC1S_RGBA = GpuCompressedPixelFormat::BC7_RGBA;
+  } else if (supportedFormats.BC3_RGBA) {
+    this->ETC1S_RGBA = GpuCompressedPixelFormat::BC3_RGBA;
+  } else if (supportedFormats.PVRTC1_4_RGBA) {
+    this->ETC1S_RGBA = GpuCompressedPixelFormat::PVRTC1_4_RGBA;
+  }
+
+  // Find the transcode target for ETC1S RGB
+  if (supportedFormats.ETC1_RGB) {
+    this->ETC1S_RGB = GpuCompressedPixelFormat::ETC1_RGB;
+  } else if (supportedFormats.BC7_RGBA) {
+    this->ETC1S_RGB = GpuCompressedPixelFormat::BC7_RGBA;
+  } else if (supportedFormats.BC1_RGB) {
+    this->ETC1S_RGB = GpuCompressedPixelFormat::BC1_RGB;
+  } else if (supportedFormats.PVRTC1_4_RGB) {
+    this->ETC1S_RGB = GpuCompressedPixelFormat::PVRTC1_4_RGB;
+  }
+
+  // Find the transcode target for ETC1S Red-Green
+  if (supportedFormats.ETC2_EAC_RG11) {
+    this->ETC1S_RG = GpuCompressedPixelFormat::ETC2_EAC_RG11;
+  } else if (supportedFormats.ETC1_RGB) {
+    this->ETC1S_RG = GpuCompressedPixelFormat::ETC1_RGB;
```

Comment:
The chart says to transcode to two ETC1 RGB textures here. I don't totally understand why, but it seems fairly clear:

> On platforms that support only ETC1 RGB, applications could transcode each ETC1S slice to its own ETC1 RGB texture and use two texture samplers simultaneously.

I'm fine with skipping the hassle of creating two textures for now and just moving on to the next option (even if that means uncompressed), but we probably shouldn't pack the two channels into one ETC1_RGB texture. We should also write an issue for remember to do the texture splitting in the future.